Cleaning spreadsheets (#8)
LaunchCodeEducation/data-analysis-curriculum: content/cleaning-spreadsheets/reading/types-of-dirty-data/_index.md

+++
title = "Types of Dirty Data"
date = 2024-01-31T13:01:06-06:00
draft = false
weight = 1
+++

There are four different types of "dirty" data. Understanding and being able to differentiate between them will help you categorize your data and decide what technique or practice would work best for cleaning a dataset. Below we will cover **missing data**, **irregular data**, **unncessary data**, and **inconsistent data**.

## Missing Data
Identifying missing data can be as straight forward as noticing empty columns or cells. If you see that a row or column is blank or void of a data entry then that can be classified as missing data.

A common example would be a survey that has questions which are optional. If the user decides not to answer the question it may result in an empty cell within your dataset. 

> There are of course ways to combat missing data that usually involves deleting the entire entry or imputing data based on existing knowledge of the dataset or a substitute value like `N/A` or `NaN`. Another approach is to use column means or regression values based on the data that you do have. 

Reviewer: ```suggestion
Analysts have ways to combat missing data. This usually involves deleting the entire entry or imputing data based on existing knowledge of the dataset or a substitute value like `N/A` or `NaN`. Another approach is to use column means or regression values based on the data that you have. 
```

Reviewer: Should the last sentence be: "Another approach is to use mathematics to help address the gaps."

> You can also revisit the place the data was first collected to see if it exists there and was lost in transit.

Reviewer: ```suggestion
You can also revisit the data storage to see if it exists there and was lost in transit.
```

{{% notice blue Example "rocket" %}}
The table below has numerous examples of missing data within the `last_name`, `email`, `employer`, `phone_number`, and `favorite_hobby` columns.

| first_name | last_name | email                    | employer            | phone_number   | favorite_hobby    |
|-|-|-|-|-|-|
| Amy        | Williams  | amorris@example.net      | Emerson Electric    | 379-012-0298   | hiking            |  
| Robert     | Marshal   |                           | Edward Jones        | 288-085-0092   | reading           |
| Kimberly   | Stephen   | ptaylor@example.net      |                     | 126-015-0765   | yoga              |
| Rachel     | Vincent   | rodneythomas@example.net | Neuroflow           |                | painting          |   
| Brian      | Salinas   | iperez@example.net       | Edward Jones        | 914-555-4392   |                   |
| Wesley     |           | stephen94@example.net    | Emerson Electtic    | 504-326-2719   | gaming       |
{{% /notice %}}

> In relation to the above examples it is important to understand whether or not the missing data was due to user error or the means of collecting the data itself. This will help when deciding on what strategy can be used to alleviate the problem.

Reviewer: ```suggestion
With the above examples, prioritize understanding whether or not the missing data was due to user error or the means of collecting the data itself. This will help when deciding on what strategy can be used to alleviate the problem.
```
